### PR TITLE
Spectrum Plot upgrades

### PR DESCRIPTION
**Pull request by guynir42**
Added tooltips to spectral lines. Added smoothing option with a customizable window size (I used moving average, nothing fancy). 
Peek 2022-01-12 14-41

### CODE REVIEW

**Comment by acrellin** [comment]
@guynir42 Can you merge master or rebase so CI can complete?

**Comment by guynir42** [comment]
@ari merged master now all is green.

**Review by profjsb — APPROVED** [review]
Looks good. Might be nicer to use Savitzky-Golay smoothing (e.g., https://github.com/mljs/savitzky-golay) instead of averaging.

**Comment by guynir42** [comment]
> Looks good. Might be nicer to use Savitzky-Golay smoothing (e.g., https://github.com/mljs/savitzky-golay) instead of averaging.

This is pretty cool. I'm not sure what the procedure is for adding a repo into the requirements for SP. 

More importantly, I think that in the JS snippet I use to run the filtering I can't really import things very easily, so I'm not sure if I can use this at all.

File: skyportal/plot.py [hunk 1]
```diff
@@ -1629,9 +1654,70 @@ def make_spectrum_layout(obj, spectra, user, device, width):
...
+    smooth_input = NumericInput(value=smooth_number)
+    smooth_checkbox = CheckboxGroup(
+        labels=["smoothing"],
+        active=[0] if smoothing else [],
+    )
+    spectrum_plots = [v for k, v in model_dict.items()]
+
+    smooth_callback = CustomJS(
+        args=dict(plots=spectrum_plots, checkbox=smooth_checkbox, window=smooth_input),
+        code="""
```
Comment (acrellin) [inline skyportal/plot.py:1671]: Why not put this in an external file like we do for other CustomJS features?

Comment (guynir42) [reply skyportal/plot.py:1671]: I haven't seen any examples of this in the code I've been working on. Honestly I didn't know that was an option...

Comment (acrellin) [reply skyportal/plot.py:1671]: https://github.com/skyportal/skyportal/blob/master/skyportal/plot.py#L746-L749

Comment (guynir42) [reply skyportal/plot.py:1671]: got it!

**Comment by profjsb** [comment]
Graying out the original data should be added (as in https://user-images.githubusercontent.com/17696889/127218623-ff547815-5b4f-45d2-9269-430600c8b984.gif). Can you regenerate the movie in the original comment above with the latest look and feel?

Closes fritz-marshal/fritz-beta-feedback#109
Closes #1266
Closes PR https://github.com/skyportal/skyportal/pull/2149

**Comment by guynir42** [comment]
@profjsb added a new gif with the functionality from Kyung's PR.

**Comment by profjsb** [comment]
Will merge once green

**Pull request merged**